Server/Controllers/MatchController.cs

```diff
@@ -213,8 +221,15 @@ public async Task<ActionResult> After([FromBody] AfterMatchRequest request)
 
             var newPlayers = request.Players
                 .Where(r => existingPlayers.All(p => p.SteamId.ToString() != r.SteamId))
-                .Select(p => new Player() { SteamId = ulong.Parse(p.SteamId) })
+                .Select(p => new Player() { SteamId = ulong.Parse(p.SteamId), Rating12v12 = 2000 })
                 .ToList();
+            var allPlayers = existingPlayers.Union(newPlayers).ToList();
+            var playerMatcher = new Dictionary<string, AfterMatchResponse.Player>();
+            foreach (var player in allPlayers)
+            {
+                if (!playerMatcher.ContainsKey(player.SteamId.ToString()))
+                    playerMatcher.Add(player.SteamId.ToString(), new AfterMatchResponse.Player { SteamId = player.SteamId, OldRating = player.Rating12v12 });
+            }
```

Comment:
I'm kind of surprised .net lets it pass, I thought value types were non-nullable by default? Well, anyway, it would be an error once nullable would be enabled globally. Just save old ratings and build `Players` value later

```suggestion

            var oldPlayerRatings = allPlayers.ToDictionary(p => p.SteamId, p => p.Rating12v12);
```